cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -0,0 +1,112 @@
+"Contentstore Views"
+
+from edx_rest_framework_extensions import permissions
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework.response import Response
+
+from common.lib.xmodule.xmodule.course_module import get_available_providers
+from contentstore.views.course import get_course_and_check_access
+from models.settings.course_metadata import CourseMetadata
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import view_auth_classes
+from xmodule.modulestore.django import modulestore
+
+
+from .serializers import ProctoredExamSettingsMainSerializer
+
+
+@api_view(['GET'])
+@authentication_classes([
+    JwtAuthentication,
+    BearerAuthenticationAllowInactiveUser,
+    SessionAuthenticationAllowInactiveUser,
+])
+@permission_classes([permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS])
```

Comment:
You don't need to use this permission class. This permission class is needed only for APIs that need to support OAuth Scopes.

Rather, you can add [this decorator](https://github.com/edx/edx-platform/blob/f9d379203e0eeddf5bbd185cd152d773af863a9a/openedx/core/lib/api/view_utils.py#L113) to your class.  It will handle both the authentication and permission classes that are needed.

Reply:
Ah, okay. The OAuth Scopes detail is what I was missing. Thank you.